source/unified-test-format/tests/invalid/entity-createRunCursorCommand.yml

```diff
@@ -0,0 +1,31 @@
+description: runCursorCommand
```

Comment:
A few issues here:

The `invalid/` directory is intended for test files that fail schema validation. If this is instead a test that yields an error/exception, it belongs in the `valid-fail` directory.

Tests should use the base filename as their description. So this would be `entity-createRunCursorCommand`; however, this test is using `runCommandCursor` instead of `createRunCursorCommand`.

The naming convention for "entity" tests is `entity-<entityType>-<subject>...`, so a more appropriate name might be `entity-commandCursor-expectResult-unsupported`; however the Unified Test Format spec doesn't define a `commandCursor` entity type. I left a separate comment about that on the spec diff.

----
Edit: I didn't see anything in the spec about `expectResult` being prohibited for `runCommandCursor`. I assume you made a typo in `operation.name` below and meant to use `createRunCursorCommand` in this file (per @dariakp's request for tests in https://github.com/mongodb/specifications/pull/1412/files#r1180718340).

This isn't something we can test. Note that no such tests exist for `createFindCursor`.

I think this file should be deleted.

Reply:
Oh, interesting, I did not realize we do not enumerate the specific operations via oneOf/anyOf in the `operation` definition in the schema. Was the motivation for that convenience or some sort of technical limitation? It makes sense now with your other comment about not bumping the schema version. I had always assumed that a driver implementing a runner to handle a particular schema version would guarantee that the runner is able to execute new tests in that schema version without any part of the runner needing changes, but it seems that changes to the test runner's entity map are not included in that guarantee, is that accurate?

Reply:
> Was the motivation for that convenience or some sort of technical limitation?

Mostly convenience. If we went the route of defining supported operations in the schema, then the next logical step would be modeling the operation arguments in the schema. That would be a monumental task and also lead to a lot of churn as APIs in other specs are updated (and extra work since figuring out the correct JSON schema syntax would easily outweigh the actual API change in another spec doc).

[Executing an Operation](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#executing-an-operation) requires test runners to error on unsupported operations and/or arguments, so I'm not too worried about things falling through the cracks.

> I had always assumed that a driver implementing a runner to handle a particular schema version would guarantee that the runner is able to execute new tests in that schema version without any part of the runner needing changes, but it seems that changes to the test runner's entity map are not included in that guarantee, is that accurate?

New entities _may_ warrant schema version changes if new syntax is introduced to `createEntities`, which is modeled. However, some entities are only created by operations, such as `createFindCursor`, which wouldn't necessarily require a schema version bump on its own. In the case of the find cursor, the change was accompanied by other changes that _did_ modify the schema.

I think what's also missing here is some design rationale about what is and is not modeled in the schema. I created [DRIVERS-2626](https://jira.mongodb.org/browse/DRIVERS-2626) to remind me to come back to this.